pyanaconda/modules/timezone/initialization.py

```diff
@@ -0,0 +1,43 @@
+#
+# Copyright (C) 2022  Red Hat, Inc.
+#
+# This copyrighted material is made available to anyone wishing to use,
+# modify, copy, or redistribute it subject to the terms and conditions of
+# the GNU General Public License v.2, or (at your option) any later version.
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY expressed or implied, including the implied warranties of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General
+# Public License for more details.  You should have received a copy of the
+# GNU General Public License along with this program; if not, write to the
+# Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA
+# 02110-1301, USA.  Any Red Hat trademarks that are incorporated in the
+# source code or documentation are not subject to the GNU General Public
+# License and may only be used or replicated with the express permission of
+# Red Hat, Inc.
+#
+from pyanaconda.modules.common.task import Task
+from pyanaconda.modules.timezone.geoloc import provider_id_to_class
+
+from pyanaconda.anaconda_loggers import get_module_logger
+log = get_module_logger(__name__)
+
+
+class GeolocationTask(Task):
+    """Run geolocation"""
+    def __init__(self, provider_id):
```

Comment:
I really think it would be useful to have the provider in the Anaconda configuration files. I could image that some products (like RHEL) might want to have special requests for the geolocation settings. Based on that, I could write a deprecation plan for dropping the provider from the boot options, so `inst.geoloc` would be just a simple boolean option in the future.